integr8ly/keycloak-operator
# INTLY-2377 adding support for browser auth flow redirect

## Motivation
JIRA: https://issues.jboss.org/browse/INTLY-2377


## What
Adding new optional field that will ensure automatic redirect into desired IDP.

## Why
For smooth login experience user should be automatically redirected into identity provider, if such provider is set via new field in realm spec - `browserRedirectorIdentityProvider`.

## How
The field has been added to KeycloakRealm.spec.
New realm reconciliation code was added.
Functions for required Keycloak APIs have been added to the client.

## Verification Steps

1. Install this version of keycloak-operator into "sso" namespace and create keycloak instance via CR. Edit: make sure to specify `spec.plugins: ["openshift4-idp"]`
2. Create "codeready" NS and install CRW operator from OperatorHub
3. Create CodeReady instance - use following CR as example (replace domain of your cluster) - https://gist.github.com/matskiv/e60251d5951d77b3fd1613329f123cad#file-codeready-cr-yml
4. Create OAuthClient - https://gist.github.com/matskiv/e60251d5951d77b3fd1613329f123cad#file-oauth_client-yml Edit: you need to update redirectURIs value to match your keycloak URL.
5. Create "openshift" realm - use following CR as example (replace domain of your cluster) - https://gist.github.com/matskiv/e60251d5951d77b3fd1613329f123cad#file-keycloak_realm-yml
6. After realm has been successfully reconciled, open CodeReady via route in "codeready" NS.
7. You should be automatically redirected to Openshift login screen

You can also try removing "browserRedirectorIdentityProvider" field from the realm spec, logout from codeready and try opening it again. In such case you should see intermediary screen where on one side will be Keycloaks login form, and on other side will be button for "OpenShift v4" IDP.

## Checklist:
